src/content/capture/video-contenthint/js/main.js

```diff
@@ -0,0 +1,151 @@
+
+var srcStream;
+var fluidStream;
+var detailedStream;
+
+var offerOptions = {
+  offerToReceiveAudio: 0,
+  offerToReceiveVideo: 1
+};
+
+function maybeCreateStream() {
+  if (srcStream) {
+    return;
+  }
+  if (srcVideo.captureStream) {
+    srcStream = srcVideo.captureStream();
+    console.log('Captured stream from srcVideo with captureStream',
+      srcStream);
+    call();
+  } else if (srcVideo.mozCaptureStream) {
+    srcStream = srcVideo.mozCaptureStream();
+    console.log('Captured stream from srcVideo with mozCaptureStream()',
+      srcStream);
+    call();
+  } else {
+    trace('captureStream() not supported');
+  }
+}
+
+// Video tag capture must be set up after video tracks are enumerated.
+srcVideo.oncanplay = maybeCreateStream;
+if (srcVideo.readyState >= 3) {  // HAVE_FUTURE_DATA
+  // Video is already ready to play, call maybeCreateStream in case oncanplay
+  // fired before we registered the event handler.
+  maybeCreateStream();
+}
+
+srcVideo.play();
+
+function setVideoTrackContentHints(stream, hint) {
+  var tracks = stream.getVideoTracks();
+  tracks.forEach(function(track) {
+    if ('contentHint' in track) {
+      track.contentHint = hint;
+      if (track.contentHint !== hint) {
+        trace('Invalid video track contentHint: \'' + hint + '\'');
+      }
+    } else {
+      trace('MediaStreamTrack contentHint attribute not supported');
+    }
+  });
+}
+
+function call() {
+  fluidStream = srcStream.clone();
+  setVideoTrackContentHints(fluidStream, 'fluid');
+  establishPC(fluidVideo, fluidStream);
+  detailedStream = srcStream.clone();
+  setVideoTrackContentHints(detailedStream, 'detailed');
+  establishPC(detailedVideo, detailedStream);
+}
+
+function establishPC(videoTag, stream) {
+  var pc1 = new RTCPeerConnection(null);
+  var pc2 = new RTCPeerConnection(null);
+  pc1.onicecandidate = function(e) {
+    onIceCandidate(pc1, pc2, e);
+  };
+  pc2.onicecandidate = function(e) {
+    onIceCandidate(pc2, pc1, e);
+  };
+  pc2.onaddstream = function(event) {
+    gotRemoteStream(videoTag, event);
+  };
+
+  pc1.addStream(stream);
+  pc1.createOffer(
+      offerOptions
+  ).then(
+    function(desc) {
+      onCreateOfferSuccess(pc1, pc2, desc);
+    },
+    onCreateSessionDescriptionError
+  );
+}
+
+function onCreateSessionDescriptionError(error) {
+  trace('Failed to create session description: ' + error.toString());
+}
+
+function onCreateOfferSuccess(pc1, pc2, desc) {
+  pc1.setLocalDescription(desc).then(
+    function() {},
+    onSetSessionDescriptionError
+  );
+  pc2.setRemoteDescription(desc).then(
+    function() {},
+    onSetSessionDescriptionError
+  );
+  // Since the 'remote' side has no media stream we need
+  // to pass in the right constraints in order for it to
+  // accept the incoming offer of audio and video.
+  pc2.createAnswer().then(
+    function(answerDesc) {
+      onCreateAnswerSuccess(pc1, pc2, answerDesc);
+    },
+    onCreateSessionDescriptionError
```

Comment:
Style preference: I prefer .catch(errorhandler) instead of the 2-argument .then, since that will cause errors inside the first .then handler to be caught too.